fuel/datasets/hdf5.py

```diff
@@ -242,8 +271,16 @@ def load_axis_labels(self):
         handle = self._file_handle
         axis_labels = {}
         for source_name in handle:
-            axis_labels[source_name] = tuple(
-                dim.label for dim in handle[source_name].dims)
+            vlen_source = (len(handle[source_name].dims) == 1
```

Comment:
Copy-paste detected: you have these conditions for a source to be considered a variable-length source in at least two places in your code. Maybe `def is_variable_length(source_name)` is worth adding to the class?